services/web-app/lib/file-cache.js

```diff
@@ -96,6 +100,40 @@ export const getResponse = (host, route, options) => {
   })
 }
 
+/**
+ * Just like `getResponse`, but before returning a GitHub response using the cache, it attempts to
+ * read the contents as a string and optimize using SVGO, returning a base64-encoded SVG.
+ * @param {string} host - GitHub API base URL
+ * @param {string} route - GitHub API request route
+ * @param {object} options - Options merged into the request route
+ * @returns {Promise<object>} GitHub API response data
+ */
+export const getSvgResponse = (host, route, options = {}) => {
+  const responseKey = slugifyRequest(host, route, options)
+  // console.log('CACHE HIT', responseKey)
+
+  return diskCache.wrap(responseKey, () => {
+    const data = _getResponse(host, route, options)
+
+    try {
+      const originalSvgString = Buffer.from(data.content, data.encoding).toString('utf8')
+      const optimizedSvgResult = optimize(originalSvgString)
+      const optimizedSvgString = optimizedSvgResult.data
+
+      return {
+        ...data,
+        content: Buffer.from(optimizedSvgString, 'utf8').toString('base64')
+      }
+    } catch (err) {
+      logging.info(
+        `Unable to optimize the SVG from ${host}/${options.owner}/${options.repo}/${options.path} with the ref ${options.ref}`
+      )
+
+      return data
+    }
+  })
+}
```

Comment:
I think we need to await the `_getresponse` otherwise the data might not have been resolved by the time the buffer is created, tested deleting the cached files and it wouldn't work for me until I added the await 
```suggestion
export const getSvgResponse = async (host, route, options = {}) => {
  const responseKey = slugifyRequest(host, route, options)
  // console.log('CACHE HIT', responseKey)

  return diskCache.wrap(responseKey, async () => {
    const data = await _getResponse(host, route, options)

    try {
      const originalSvgString = Buffer.from(data.content, data.encoding).toString('utf8')
      const optimizedSvgResult = optimize(originalSvgString)
      const optimizedSvgString = optimizedSvgResult.data

      return {
        ...data,
        content: Buffer.from(optimizedSvgString, 'utf8').toString('base64')
      }
    } catch (err) {
      logging.info(
        `Unable to optimize the SVG from ${host}/${options.owner}/${options.repo}/${options.path} with the ref ${options.ref}`
      )

      return data
    }
  })
}
```

Reply:
I originally had an await there too, but Sonarlint threw an error, saying the await was unnecessary?

Reply:
probably something to do with the return type of `_getRepsonse` : 
```
/**
 * Internal function that proxies GitHub requests.
 * @param {string} host - GitHub API base URL
 * @param {string} route - GitHub API request route
 * @param {object} options - Options merged into the request route
 * @returns {object} GitHub API response data
 */
```
try making it a promise instead: 
```
/**
 * Internal function that proxies GitHub requests.
 * @param {string} host - GitHub API base URL
 * @param {string} route - GitHub API request route
 * @param {object} options - Options merged into the request route
 * @returns {Promise<object>} GitHub API response data
 */
```
I can help troubleshoot if that still doesn't fix it

Reply:
Ah. I didn't even think of invalid JSDoc throwing off SonarLint but that makes sense. I'll make another pass over this to make sure.